src/vconsole/vconsole-setup.c

```diff
@@ -54,6 +54,55 @@ static bool is_vconsole(int fd) {
         return ioctl(fd, TIOCLINUX, data) >= 0;
 }
 
+static bool is_allocated(int fd, int idx) {
+        struct vt_stat vcs = {};
+        char vcname[strlen("/dev/vcs") + DECIMAL_STR_MAX(int)];
+
+        if (fd >= 0) {
+                if (ioctl(fd, VT_GETSTATE, &vcs) < 0) {
+                        (void) log_warning_errno(errno, "VT_GETSTATE failed: %m");
+                        return false;
+                }
+                idx = vcs.v_active;
+        }
+
+        xsprintf(vcname, "/dev/vcs%i", idx);
+        if (access(vcname, F_OK) < 0)
+                return false;
+        return true;
+}
```

Comment:
after all i figure just leaving this as "bool" is OK.

if you would like to fix this anyway, then my suggestion would be to return "extended bools", i.e. return an "int", where < 0 is an error (negtive errno), == 0 is "false" and > 0 is "true". We do that a quite a few areas. It's a clean extension of a C bool, and casts implicitly to it and from it, given that C99 bools are in most ways practically ints anyway... And callers have the choice if they want to handle the error as error, or just consider an error either a "true" or as "false" simply be comparing the return value with "!= 0" or with "> 0"...